backstory/src/test/java/com/google/sps/perspective/PerspectiveAPIClientTest.java

```diff
@@ -0,0 +1,191 @@
+import org.mockito.stubbing.Answer;
+
+/** Quality tests for PerspectiveAPIClient */
+@RunWith(JUnit4.class)
+public final class PerspectiveAPIClientTest {
+
+  /** an instance of the PerspectiveAPIClient to use throughout our tests */
+  private static PerspectiveAPIClient client;
+  /** a mock PerspectiveAPI object to be used as the input api throughout the tests */
+  private static PerspectiveAPI mockAPI;
+  /** the array of attribute types that we want scores back for */
+  private static final AttributeType[] desiredTypes = { AttributeType.PROFANITY, AttributeType.TOXICITY, AttributeType.UNSUBSTANTIAL };;
+
+
+  @Before 
+  public void setUp() {
+    // will set up an API that only returns -1
+    mockAPI = createMockAPI(new HashMap<AttributeType, Float>()); 
+
+    client = new PerspectiveAPIClient(mockAPI);
+  }
+
+  /**
+   * Construct a PerspectiveAPIClient with a null PerspectiveAPI to ensure 
+   * that an IllegalArgumentException will be thrown.
+   */
+  @Test (expected = IllegalArgumentException.class)
+  public void nullAPIInput() {
+    client = new PerspectiveAPIClient(null); 
+  }
+
+  /**
+   * Call analyze() with a null Attribute[] to ensure 
+   * that an IllegalArgumentException will be thrown.
+   */
+  @Test (expected = IllegalArgumentException.class)
+  public void nullAttributeArrayInput() {
+    client.analyze(null, "foo");
+  }
+
+  /**
+   * Call analyze() with a null text input to ensure 
+   * that an IllegalArgumentException will be thrown.
+   */
+  @Test (expected = IllegalArgumentException.class)
+  public void nullTextInput() {
+    client.analyze(desiredTypes, null);
+  }
+
+  /**
+   * Call analyze() with an empty text input to ensure 
+   * that an IllegalArgumentException will be thrown.
+   */
+  @Test (expected = IllegalArgumentException.class)
+  public void emptyTextInput() {
+    client.analyze(desiredTypes, "");
+  }
+
+  /**
+   * Check that analyze() returns a PerspectiveValues object
+   * with a map of the correct size.
+   */
+  @Test
+  public void checkSizeOfMap() {
+    PerspectiveValues values = client.analyze(desiredTypes, "foo");
+
+    Assert.assertEquals(desiredTypes.length, values.getAttributeTypesToScores().size());
+  }
+
+  /**
+   * Check that analyze() returns a PerspectiveValues
+   * object with the same text as passed in.
+   */
+  @Test
+  public void checkTextSame() {
+    PerspectiveValues values = client.analyze(desiredTypes, "foo");
+
+    Assert.assertEquals("foo", values.getText());
+  }
+
+  /**
+   * Check that analyze() returns a PerspectiveValues
+   * object with the right scores.
+   */
+  @Test
+  public void checkRightScores() {
+    // a Map to be used as the desired scores for the mock API to return 
+    Map desiredScores = new HashMap<AttributeType, Float>();
+
+    // add each of the desiredTypes and a random float between 0 & 1 as their score 
+    for (AttributeType type: desiredTypes) {
+      Random rand = new Random();
```

Comment:
`rand` -> `random`

```
do not abbreviate by deleting letters within a word
```
https://google.github.io/styleguide/jsguide.html#naming-rules-common-to-all-identifiers

Reply:
Thanks for pointing that out!

Reply:
Done